### PR TITLE
Improve docs of softmax

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
This is PR for improving docs of functions/links. Related issue: #2182

Modified functions/links:
- functions/activation/softmax

### CODE REVIEW

**Comment by keisuke-umezawa** [comment]
The travis-ci tests will not be passed, because autopep8 generates the diffs in the files not related to this PR.
This phenomenon also occurs at my local master branch and the other's [PR build](https://travis-ci.org/pfnet/chainer/builds/207796476).
 
Should I need to some setup?

```
~/dev/chainer]
$ autopep8 -r . --global-config .pep8 --diff | tee check_autopep8
--- original/./chainer/__init__.py
+++ fixed/./chainer/__init__.py
@@ -78,6 +78,7 @@
         thread_local.function_hooks = collections.OrderedDict()
     return thread_local.function_hooks
 
+
 _debug = False
 
 
@@ -127,6 +128,7 @@
     def __exit__(self, *_):
         set_debug(self._old)
 
+
 basic_math.install_variable_arithmetics()
 array.get_item.install_variable_get_item()
 
--- original/./chainer/utils/conv_nd_kernel.py
+++ fixed/./chainer/utils/conv_nd_kernel.py
@@ -164,6 +164,7 @@
     @chainer.cuda.memoize()
     def generate(ndim):
         return _im2col_nd_kernel._generate(ndim)
+
 
 _im2col_nd_kernel = Im2colNDKernel()
 
@@ -289,4 +290,5 @@
     def generate(ndim):
         return _col2im_nd_kernel._generate(ndim)
 
+
 _col2im_nd_kernel = Col2imNDKernel()

... cont'd
```

**Comment by keisuke-umezawa** [comment]
The cause may be the version of autopep8.
My build and  [PR build](https://travis-ci.org/pfnet/chainer/builds/207796476) use autopep8 of 1.3.0.
But the builds before use autopep8 of 1.2.4.
So, there are some updates from 1.2.4 to 1.3.0 and they generate some diff for other files.

**Comment by keisuke-umezawa** [comment]
remove WIP. The latest commit fails on Travis because of an autopep8 problem that will be fixed in #2363. I will kick Travis again, after it will be merged to the master.

File: chainer/functions/activation/softmax.py [hunk 1]
```diff
@@ -74,22 +74,38 @@ def backward(self, x, gy):
...
+    For each input array :math:`x`, it computes the function :math:`f(x)`
+    defined as
+    :math:`f(x)={\\exp(x) \\over \\sum_{x_d} \\exp(x_d)}`.
 
     Args:
-        x (~chainer.Variable): Input variable.
+        X (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variable.
+            A 2d (N, D) :math:`((s_1^1, ..., s_D^1), ..., (s_1^N, ..., s_D^N))`
```
Comment (mitmul) [inline chainer/functions/activation/softmax.py]: `softmax` can also be applied to more than 2-dimensional tensor, e.g., a minibatch of images shaped like `(N, C, H, W)`. In such case, the output will be `(N, C, H, W)`-shaped tensor and the summation over the channel axis will be always 1. In short, the output is not always "2d (N, D)", so I'd like you to fix this part ;)

File: chainer/functions/activation/softmax.py [hunk 2]
```diff
@@ -74,22 +74,38 @@ def backward(self, x, gy):
...
+
+    .. admonition:: Example
+
+        >>> x = np.array([[0, 1, 2], [0, 2, 4]], 'f')
+        >>> x
+        array([[ 0.,  1.,  2.],
+               [ 0.,  2.,  4.]], dtype=float32)
+        >>> F.softmax(x).data
+        array([[ 0.09003057,  0.24472848,  0.66524094],
+               [ 0.01587624,  0.11731043,  0.86681336]], dtype=float32)
```
Comment (mitmul) [inline chainer/functions/activation/softmax.py]: How about showing that the summation over the second axis is always 1? For example,
```
>>> y = F.softmax(x)
>>> y.data
array([[ 0.09003057,  0.24472848,  0.66524094],
       [ 0.01587624,  0.11731043,  0.86681336]], dtype=float32)
>>> F.sum(y, axis=1).data
array([ 1.,  1.], dtype=float32)
```

**Comment by keisuke-umezawa** [comment]
put doctest result
```
...

Document: reference/functions
-----------------------------
1 items passed all tests:
  89 tests in default
89 tests in 1 items.
89 passed and 0 failed.
Test passed.

...

Doctest summary
===============
  308 tests
    8 failures in tests
    0 failures in setup code
    0 failures in cleanup code
build finished with problems.
make: *** [doctest] Error 1
```

**Comment by keisuke-umezawa** [comment]
@mitmul 
I rebased this branch against master. Please rerun the Jenkins test.

**Comment by keisuke-umezawa** [comment]
@mitmul 
I had some comment of log-softmax ( #2411 ) and reflected this PR.
Please reveiw this PR.

**Comment by mitmul** [comment]
@keisuke-umezawa Could you resolve the conflicts? Now it's pointed to _v2 branch. Sorry for late reaction.

**Comment by keisuke-umezawa** [comment]
@mitmul
I resolved the conflict. plz merge this.

**Comment by mitmul** [comment]
Jenkins, test this please!

**Comment by mitmul** [comment]
LGRWKJG
